backend/src/apiserver/storage/system_info_store.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	sq "github.com/Masterminds/squirrel"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+type SystemInfoStoreInterface interface {
+	IsSampleLoaded() (bool, error)
+	MarkSampleLoaded() error
+}
+
+type SystemInfoStore struct {
+	db *DB
+}
+
+func (s *SystemInfoStore) InitializeSystemInfoTable() error {
+	getSystemInfoSql, getSystemInfoArgs, err := sq.Select("*").From("system_infos").ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err, "Error creating query to get system info.")
+	}
+	tx, err := s.db.Begin()
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to create a new transaction to initialize system info.")
+	}
+
+	rows, err := tx.Query(getSystemInfoSql, getSystemInfoArgs...)
+	if err != nil {
+		tx.Rollback()
+		return util.NewInternalServerError(err, "Failed to get load sample status")
+	}
+
+	// The table is not initialized
+	if !rows.Next() {
+		sql, args, err := sq.
+			Insert("system_infos").
+			SetMap(sq.Eq{"IsSampleLoaded": false}).
```

Comment:
it's actually only storing the status of the database. so i rename it to db_status_store. it would make more sense to be stored inside the db itself.